tuning/trainercontroller/operations/operation.py

```diff
@@ -0,0 +1,45 @@
+# Standard
+import abc
+import re
+import inspect
+
+class Operation(metaclass=abc.ABCMeta):
+    """Base class for operations"""
+
+    def __init__(self):
+        """Initializes the HuggingFace controls. In this init, we follow the convention that
+        every action should preceed with prefix `should_`. If so, it is treated as a valid
+        action.
+        """
+        self.valid_actions = {}       
+        for action_name, action_method in inspect.getmembers(self, predicate=inspect.ismethod):
+            if re.search(r'^should_', action_name) is not None:
+                self.valid_actions[action_name] = action_method
+
+    def validate(self, action: str) -> bool:
+        """Validates the action by checking if it valid action or not.
+
+        Args:
+            action: str. String depicting the action.
+
+        Returns:
+            bool. Indicates True if valid. If not, returns False.
+        """
+        if action not in self.valid_actions:
+            return False
```

Comment:
```suggestion
        return action in self.valid_actions
```

Reply:
@seshapad Let me know if I have missed something, the function would never be returning True with the current code isn't it?

Reply:
Addressed